Sources/Sentry/SentrySessionReplay.m

```diff
@@ -0,0 +1,264 @@
+
+@implementation SentrySessionReplay {
+    UIView *_rootView;
+    BOOL _processingScreenshot;
+    CADisplayLink *_displayLink;
+    NSDate *_lastScreenShot;
+    NSDate *_videoSegmentStart;
+    NSURL *_urlToCache;
+    NSDate *_sessionStart;
+    SentryReplayOptions *_replayOptions;
+    SentryOnDemandReplay *_replayMaker;
+    SentryId *sessionReplayId;
+    NSMutableArray<UIImage *> *imageCollection;
+    int _currentSegmentId;
+    BOOL _isFullSession;
+}
+
+- (instancetype)initWithSettings:(SentryReplayOptions *)replayOptions
+{
+    if (self = [super init]) {
+        _replayOptions = replayOptions;
+    }
+    return self;
+}
+
+- (SentryCurrentDateProvider *)dateProvider
+{
+    return SentryDependencyContainer.sharedInstance.dateProvider;
+}
+
+- (void)start:(UIView *)rootView fullSession:(BOOL)full
+{
+    if (rootView == nil) {
+        SENTRY_LOG_DEBUG(@"rootView cannot be nil. Session replay will not be recorded.");
+        return;
+    }
+
+    @synchronized(self) {
+        if (_displayLink == nil) {
+            _displayLink = [CADisplayLink displayLinkWithTarget:self selector:@selector(newFrame:)];
+            [_displayLink addToRunLoop:[NSRunLoop mainRunLoop] forMode:NSRunLoopCommonModes];
+        } else {
+            // Session display is already running.
+            return;
+        }
+
+        _rootView = rootView;
+        _lastScreenShot = [[NSDate alloc] init];
+        _videoSegmentStart = nil;
+        _sessionStart = _lastScreenShot;
+        _currentSegmentId = 0;
+
+        NSURL *docs = [NSURL
+            fileURLWithPath:[SentryDependencyContainer.sharedInstance.fileManager sentryPath]];
+        docs = [docs URLByAppendingPathComponent:SENTRY_REPLAY_FOLDER];
+
+        NSString *currentSession = [NSUUID UUID].UUIDString;
+        _urlToCache = [docs URLByAppendingPathComponent:currentSession];
+
+        if (![NSFileManager.defaultManager fileExistsAtPath:_urlToCache.path]) {
+            [NSFileManager.defaultManager createDirectoryAtURL:_urlToCache
+                                   withIntermediateDirectories:YES
+                                                    attributes:nil
+                                                         error:nil];
+        }
+
+        _replayMaker = [[SentryOnDemandReplay alloc] initWithOutputPath:_urlToCache.path];
+        _replayMaker.bitRate = _replayOptions.replayBitRate;
+        _replayMaker.cacheMaxSize = (NSInteger)(full ? _replayOptions.sessionSegmentDuration
+                                                     : _replayOptions.errorReplayDuration);
+        imageCollection = [NSMutableArray array];
+
+        NSLog(@"Recording session to %@", _urlToCache);
+
+        _isFullSession = full;
+        if (full) {
+            sessionReplayId = [[SentryId alloc] init];
+        }
+    }
+}
+
+- (void)stop
+{
+    [_displayLink invalidate];
+    _displayLink = nil;
+}
+
+- (void)replayForEvent:(SentryEvent *)event;
+{
+    if (_isFullSession) {
```

Comment:
I think you wanna get rid of this one, because it will return-early  and won't get to the one below (line 114)

Reply:
This is correct, the rest of the function it isn't.
The DSC would have the replayId already if the full session replay started.

Reply:
ok, but the `sample(errorSampleRate)` call is still missing here I believe

Reply:
Oh, yes. I had to revert back from Swift to Objc and it was lost in the process. Thanks!